agent/controller/agent.go

```diff
@@ -193,6 +193,24 @@ func (a *Agent) processSessionOpen(pkt *pb.Packet) {
 		}
 	}
 
+	if a.connStore.Get(dlpProviderKey) == nil {
+		if dlpProvider, ok := pkt.Spec[pb.SpecAgentDlpProvider]; ok {
+			a.connStore.Set(dlpProviderKey, dlpProvider)
+		}
+	}
+
+	if a.connStore.Get(msPresidioAnonymizerURLKey) == nil {
+		if anonymizerURL, ok := pkt.Spec[pb.SpecAgentMSPresidioAnonymizerURL]; ok {
+			a.connStore.Set(msPresidioAnonymizerURLKey, anonymizerURL)
+		}
+	}
+
+	if a.connStore.Get(msPresidioAnalyzerURLKey) == nil {
+		if analyzerURL, ok := pkt.Spec[pb.SpecAgentMSPresidioAnalyzerURL]; ok {
+			a.connStore.Set(msPresidioAnalyzerURLKey, analyzerURL)
+		}
+	}
```

Comment:
TIP: you can save them in a single struct using a single key, but it's ok to leave it like for now.